app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,24 @@
+class CommentsController < ApplicationController
+  before_action :logged_in_user, only: [:create]
+
+  def create
+    post = Post.find(params[:comment][:post_id])
+    if params[:comment][:comment_id].present?
+      parent = Comment.find(params[:comment][:comment_id])
+    else
+      parent = post
+    end
+    comment = parent.comments.build
+    if (comment.update_attributes(comment_params))
+      comment.save
+    else
+      flash[:danger] = "Comment may not be empty."
+    end
+    redirect_to post
+  end
+
+  def comment_params
+    params.require(:comment).permit(:content, :user_id)
```

Comment:
`user_id` is being passed in as a hidden field from the show view for a given post (containing all comments). It refers to the current user who is creating the comment.